crates/kitsune-http-client/src/lib.rs

```diff
@@ -251,7 +237,15 @@ impl Client {
     ///
     /// - The inner client service isn't ready
     /// - The request failed
-    pub async fn execute(&self, req: Request<Body>) -> Result<Response> {
+    pub async fn execute<B>(&self, req: Request<B>) -> Result<Response>
+    where
+        B: HttpBody<Data = Bytes> + Send + Sync + 'static,
+        B::Error: Into<BoxError> + 'static,
+    {
+        let (parts, body) = req.into_parts();
+        let body = BoxBody::new(body.map_err(Into::into));
```

Comment:
I believe the type-erasure can be done in a separate function to reduce the monomorphisation cost.